build.psm1

```diff
@@ -188,6 +190,12 @@ function Start-PSBuild {
     # verify we have all tools in place to do the build
     $precheck = precheck 'dotnet' "Build dependency 'dotnet' not found in PATH. Run Start-PSBootstrap. Also see: https://dotnet.github.io/getting-started/"
 
+    $dotnetIntalledVersion = (dotnet --version)
+    If ( $dotnetIntalledVersion -ne $dotnetRequiredVersion ) {
+        Write-Warning ".Net Core version mismatch: installed version is $dotnetIntalledVersion, required version is $dotnetRequiredVersion.`nRun Start-PSBootstrap and Start-PSBuild -Clean -Restore.`nAlso see: https://dotnet.github.io/getting-started/"
```

Comment:
The message should also print out the path to `dotnet` - a user might have multiple versions installed and is picking up the wrong one.